packages/cw721/src/execute.rs

```diff
@@ -0,0 +1,861 @@
+        _env: &Env,
+        info: &MessageInfo,
+        operator: String,
+    ) -> Result<Response<TCustomResponseMsg>, Cw721ContractError> {
+        let operator_addr = deps.api.addr_validate(&operator)?;
+        let config = Cw721Config::<
+            TNftMetadataExtension,
+            TNftMetadataExtensionMsg,
+            TCollectionMetadataExtension,
+            TCollectionMetadataExtensionMsg,
+            TCustomResponseMsg,
+        >::default();
+        config
+            .operators
+            .remove(deps.storage, (&info.sender, &operator_addr));
+
+        Ok(Response::new()
+            .add_attribute("action", "revoke_all")
+            .add_attribute("sender", info.sender.to_string())
+            .add_attribute("operator", operator))
+    }
+
+    fn burn_nft(
+        &self,
+        deps: DepsMut,
+        env: &Env,
+        info: &MessageInfo,
+        token_id: String,
+    ) -> Result<Response<TCustomResponseMsg>, Cw721ContractError> {
+        let config = Cw721Config::<
+            TNftMetadataExtension,
+            TNftMetadataExtensionMsg,
+            TCollectionMetadataExtension,
+            TCollectionMetadataExtensionMsg,
+            TCustomResponseMsg,
+        >::default();
+        let token = config.nft_info.load(deps.storage, &token_id)?;
+        check_can_send(deps.as_ref(), env, info, &token)?;
+
+        config.nft_info.remove(deps.storage, &token_id)?;
+        config.decrement_tokens(deps.storage)?;
+
+        Ok(Response::new()
+            .add_attribute("action", "burn")
+            .add_attribute("sender", info.sender.to_string())
+            .add_attribute("token_id", token_id))
+    }
+
+    // ------- opionated cw721 functions -------
+    fn initialize_creator(
+        &self,
+        storage: &mut dyn Storage,
+        api: &dyn Api,
+        creator: Option<&str>,
+    ) -> StdResult<Ownership<Addr>> {
+        CREATOR.initialize_owner(storage, api, creator)
+    }
+
+    fn initialize_minter(
+        &self,
+        storage: &mut dyn Storage,
+        api: &dyn Api,
+        minter: Option<&str>,
+    ) -> StdResult<Ownership<Addr>> {
+        MINTER.initialize_owner(storage, api, minter)
+    }
+
+    fn update_collection_metadata(
+        &self,
+        deps: DepsMut,
+        info: &MessageInfo,
+        env: &Env,
+        msg: CollectionMetadataMsg<TCollectionMetadataExtensionMsg>,
+    ) -> Result<Response<TCustomResponseMsg>, Cw721ContractError> {
+        let config = Cw721Config::<
+            TNftMetadataExtension,
+            TNftMetadataExtensionMsg,
+            TCollectionMetadataExtension,
+            TCollectionMetadataExtensionMsg,
+            TCustomResponseMsg,
+        >::default();
+        let current = config.collection_metadata.load(deps.storage)?;
+        let collection_metadata = msg.create(deps.as_ref(), env, info, Some(&current))?;
+        config
+            .collection_metadata
+            .save(deps.storage, &collection_metadata)?;
```

Comment:
... where in `update_collection_metadata` where on msg, a single line creates a new, updated state:

`let collection_metadata = msg.create(deps.as_ref(), env, info, Some(&current))?;`

And `create()` in `StateFactory` does a deep creation and validation: ...